allennlp/training/metrics.py

```diff
@@ -288,3 +290,196 @@ def reset(self):
         self._true_negatives = 0.0
         self._false_positives = 0.0
         self._false_negatives = 0.0
+
+
+@Metric.register("conll_span_f1")
+class ConllSpanBasedF1Measure(Metric):
+    """
+    The Conll SRL metrics are based on exact span matching. This metric
+    implements span-based precision and recall metrics for a BIO tagging
+    scheme. It will produce precision, recall and F1 measures per tag, as
+    well as overall statistics. Note that the implementation of this metric
+    is not exactly the same as the perl script used to evaluate the CONLL 2005
+    data - particularly, it does not consider continuations or reference spans
+    as constituents of the original span. However, it is a close proxy, which
+    can be helpful for judging model peformance during training.
+    """
+    def __init__(self,
+                 label_vocabulary: Dict[int, str],
+                 ignore_classes: List[int] = None) -> None:
+        """
+        Parameters
+        ----------
+        label_vocabulary : Dict[int, str], required.
+            A mapping from integer class labels to string labels. This metric
+            assumes that a BIO format is used in which the labels are of the format:
+            ["B-LABEL", "I-LABEL"].
+        ignore_classes : List[int], optional.
```

Comment:
Because you have the vocabulary, would it make sense to pass these in as strings?  Oh, I guess you're just taking the `Dict[int, str]`, so it'd be obnoxious to find the strings...  Maybe this is ok.  Makes me wonder if we should have designed the `Vocab` differently, so each namespace is a class, and we could have just taken that here, and what is now the `Vocabulary` object is just a dictionary mapping strings to these namespace vocab objects...